File: Applied Course/4.Problem Solving/1.Problems on Array/14.Increasing Triplet Subsequence.py
# https://leetcode.com/problems/increasing-triplet-subsequence
def findTriplet(arr,n):
    if(n<3):
        return False
    mid,low=-1,-1
    min_index=0
    for i in range(n):
        if(arr[i]<=arr[min_index]):
            min_index=i
        elif(mid==-1):
            low=min_index
            mid=i
        elif(arr[i] <= arr[mid]):
            low=min_index
            mid=i
        else:
            return arr[low],arr[mid],arr[i]

if __name__=='__main__':
    arr=[5, 4, 3, 7, 6, 1, 9]
    print(findTriplet(arr,len(arr)))

File: Applied Course/4.Problem Solving/1.Problems on Array/14.Increasing Triplet Subsequence.cpp
#include<bits/stdc++.h>
using namespace std;
// Logic-1
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int n=nums.size();
        for(int i=0;i<n;i++)
        {
            int lmin=INT_MAX,rmax=INT_MIN;
            for(int j=0;j<i;j++)
                if(nums[j]<nums[i])
                {
                    lmin=nums[j];
                    break;
                }
            for(int j=i+1;j<n;j++)
                if(nums[i]<nums[j])
                {
                    rmax=nums[j];
                    break;
                }
            if(lmin<nums[i] && nums[i]<rmax)
                return true;
        }
        return false;
    }
};
// Logic-2
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int n=nums.size();
        int min_index=0,low=-1,mid=-1;
        for(int i=1;i<n;i++)
        {
            if(nums[i]<=nums[min_index])
                min_index=i;
            else if(mid==-1)
            {
                low=min_index;
                mid=i;
            }
            else if(nums[i]<=nums[mid])
            {
                low=min_index;
                mid=i;
            }
            else
                return true;
        }
        return false;
    }
};
int main()
{
    // https://leetcode.com/problems/increasing-triplet-subsequence/
    return 0;
}